tools/gfx-unit-test/link-time-constant-array-size-main.slang

```diff
@@ -0,0 +1,12 @@
+extern static const int N;
+
+struct S { int xs[N]; }
```

Comment:
I think the test is too simplistic and missing important coverage.

We should call the reflection API explicitly and inspect the offset and sizes of relevant parameters/fields.


The type should contain additional fields following link time sized arrays, and we need to check the following field has the right offset when querying with a concrete program layout and had invalid offset (not 0) when querying without a concrete layout.


We should also test with types that contains resources, and nested link time sized arrays.